src/Common/src/Common.Security/PemConfigureCertificateOptions.cs

```diff
@@ -60,71 +55,50 @@ public void Configure(string name, CertificateOptions options)
+            options.IssuerChain = certChain
+               .Skip(1)
+               .Select(c => new X509Certificate2(c.GetRawCertData()))
+               .ToList();
         }
 
         public void Configure(CertificateOptions options)
         {
-            Configure(Options.DefaultName, options);
+            Configure(Microsoft.Extensions.Options.Options.DefaultName, options);
         }
 
-        internal byte[] CreatePfxContainer(X509Certificate cert, AsymmetricCipherKeyPair keys)
+        private static RSA ReadKeyFromString(string pemContents)
         {
-            var certEntry = new X509CertificateEntry(cert);
+            const string RsaPrivateKeyHeader = "-----BEGIN RSA PRIVATE KEY-----";
+            const string RsaPrivateKeyFooter = "-----END RSA PRIVATE KEY-----";
 
-            var pkcs12Store = new Pkcs12StoreBuilder()
-                .SetUseDerEncoding(true)
-                .Build();
-            var keyEntry = new AsymmetricKeyEntry(keys.Private);
-            pkcs12Store.SetKeyEntry("ServerInstance", keyEntry, new X509CertificateEntry[] { certEntry });
-
-            using (MemoryStream stream = new MemoryStream())
-            {
-                pkcs12Store.Save(stream, null, new SecureRandom());
-                var bytes = stream.ToArray();
-                return Pkcs12Utilities.ConvertToDefiniteLength(bytes);
-            }
-        }
-
-        internal AsymmetricCipherKeyPair ReadKeys(byte[] keyBytes)
-        {
-            try
-            {
-                using (var reader = new StreamReader(new MemoryStream(keyBytes)))
-                {
-                    return new PemReader(reader).ReadObject() as AsymmetricCipherKeyPair;
-                }
-            }
-            catch (Exception e)
+            if (pemContents.StartsWith(RsaPrivateKeyHeader))
             {
-                _logger?.LogError(e, "Unable to read PEM encoded keys");
-            }
+                var endIdx = pemContents.IndexOf(
+                    RsaPrivateKeyFooter,
+                    RsaPrivateKeyHeader.Length,
+                    StringComparison.Ordinal);
 
-            return null;
-        }
+                var base64 = pemContents[RsaPrivateKeyHeader.Length..endIdx];
 
-        internal X509Certificate ReadCertificate(byte[] certBytes)
-        {
-            try
-            {
-                using (var reader = new StreamReader(new MemoryStream(certBytes)))
-                {
-                    return new PemReader(reader).ReadObject() as X509Certificate;
-                }
-            }
-            catch (Exception e)
-            {
-                _logger?.LogError(e, "Unable to read PEM encoded certificate");
+                var der = Convert.FromBase64String(base64);
+                var rsa = RSA.Create();
+                rsa.ImportRSAPrivateKey(der, out _);
+                return rsa;
             }
 
-            return null;
+            // "BEGIN PRIVATE KEY" (ImportPkcs8PrivateKey),
```

Comment:
Could we delete this (since if we did do this - at least we would use a different name for method)?